pkg/oc/cli/cmd/image/append/append.go

```diff
@@ -0,0 +1,635 @@
+package append
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime"
+	"time"
+
+	units "github.com/docker/go-units"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/docker/distribution"
+	distributioncontext "github.com/docker/distribution/context"
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/client"
+	digest "github.com/opencontainers/go-digest"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/image/apis/image/docker10"
+	imagereference "github.com/openshift/origin/pkg/image/apis/image/reference"
+	"github.com/openshift/origin/pkg/image/dockerlayer"
+	"github.com/openshift/origin/pkg/image/dockerlayer/add"
+	"github.com/openshift/origin/pkg/image/registryclient"
+	"github.com/openshift/origin/pkg/image/registryclient/dockercredentials"
+)
+
+var (
+	desc = templates.LongDesc(`
+		Add layers to Docker images
+
+		Modifies an existing image by adding layers or changing configuration and then pushes that
+		image to a remote registry. Any inherited layers are streamed from registry to registry 
+		without being stored locally. The default docker credentials are used for authenticating 
+		to the registries.
+
+		Layers may be provided as arguments to the command and must each be a gzipped tar archive
+		representing a filesystem overlay to the inherited images. The archive may contain a "whiteout"
+		file (the prefix '.wh.' and the filename) which will hide files in the lower layers. All
```

Comment:
is this your convention or docker's?  is there an escape mechanism if i really want to prefix a file with `.wh`?